### PR TITLE
refactor(lodash): replace "defaults" with pure alternative

### PR DESCRIPTION
**Pull request by Haroenv**
The original lodash tests don't check that it also modifies the target, and as far as I can tell, we don't rely on that behaviour.

### CODE REVIEW

File: test/spec/functions/defaultsPure.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+it('should overwrite `undefined` values', function() {
+  var actual = defaults({'a': undefined}, {'a': 1});
+  expect(actual.a).toBe(1);
+});
+
+it('should assign `undefined` values', function() {
+  var source = {'a': undefined, 'b': 1};
+  var actual = defaults({}, source);
+
+  expect(actual).toEqual({'a': undefined, 'b': 1});
```
Comment (eunjae-lee) [inline test/spec/functions/defaultsPure.js]: Shouldn't this be like this?
```js
expect(actual).toEqual({'b': 1});
```

Comment (Haroenv) [reply test/spec/functions/defaultsPure.js]: this test was copied from lodash, but explicitly putting undefined on an object still matches toEqual I think

Comment (eunjae-lee) [reply test/spec/functions/defaultsPure.js]: Ah okay. I just tested and this passes.

```js
expect({ b: 1 }).toEqual({ a: undefined, b: 1 });
```

File: src/functions/defaultsPure.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+
```
Comment (Haroenv) [inline src/functions/defaultsPure.js]:
```suggestion
```

**Pull request merged**